spi.c

```diff
@@ -0,0 +1,103 @@
+/* 
+  Not part of GRBL, KeyMe specific
+*/
+
+#include "spi.h"
+
+#define SPI_TRANSFER_FLAG (SPSR & (1<<SPIF))
+
+
+//Setup spi
+void spi_init()
+{
+
+  DDRB = ((1<<DDB2)|(1<<DDB1) | (1 << DDB0)); //spi pins on port b MOSI SCK,SS outputs
+
+  SPCR = ((1<<SPE)|(1<<MSTR)|(1<<SPR0));//|(1<<CPOL)|(1<<CPHA));  // SPI enable, Master, f/16
+
+  //Configure SCS Pins as outputs
+  bit_true(SCS_DDR_PORT, 1 << SCS_XTABLE_DDR_PIN); 
+  bit_true(SCS_DDR_PORT, 1 << SCS_YTABLE_DDR_PIN); 
+  bit_true(SCS_DDR_PORT, 1 << SCS_GRIPPER_DDR_PIN); 
+  bit_true(SCS_DDR_PORT, 1 << SCS_CAROUSEL_DDR_PIN); 
+
+  //Set SCS to low for all steppers
+  bit_false(SCS_PORT, 1 << SCS_XTABLE_PIN);
+  bit_false(SCS_PORT, 1 << SCS_YTABLE_PIN);
+  bit_false(SCS_PORT, 1 << SCS_GRIPPER_PIN);
+  bit_false(SCS_PORT, 1 << SCS_CAROUSEL_PIN);
+
```

Comment:
All of these bit_true/bit_false calls are unnecessary, you can just create a mask with the set/clear bits and use the standard `|=  &= ^` operator combos.